lib/doctor/DiagnosticInfoBuilder.ts

```diff
@@ -0,0 +1,206 @@
+import { getProjectConfig } from '../projects';
+import { fetchProject } from '@hubspot/local-dev-lib/api/projects';
+import path from 'path';
+import pkg from '../../package.json';
+import { logger } from '@hubspot/local-dev-lib/logger';
+import { Environment } from '@hubspot/local-dev-lib/types/Config';
+import {
+  AccessToken,
+  AccountType,
+  AuthType,
+} from '@hubspot/local-dev-lib/types/Accounts';
+import { Project } from '@hubspot/local-dev-lib/types/Project';
+import { getAccountId } from '../commonOpts';
+import { getAccountConfig, getConfigPath } from '@hubspot/local-dev-lib/config';
+import { getAccessToken } from '@hubspot/local-dev-lib/personalAccessKey';
+import { walk } from '@hubspot/local-dev-lib/fs';
+import util from 'util';
+import { exec as execAsync } from 'child_process';
+import process from 'process';
+
+export type ProjectConfig = Awaited<ReturnType<typeof getProjectConfig>>;
+
+interface FilesInfo {
+  files: string[];
+  configFiles: string[];
+  packageFiles: string[];
+  packageLockFiles: string[];
+  envFiles: string[];
+  jsonFiles: string[];
+}
+
+export interface DiagnosticInfo extends FilesInfo {
+  path?: string;
+  versions: { '@hubspot/cli': string; node: string; npm: string | null };
+  config: string | null;
+  project: {
+    details?: Project;
+    config?: ProjectConfig;
+  };
+  arch: typeof process.arch;
+  platform: typeof process.platform;
+  account: {
+    name?: string;
+    accountId?: number | null;
+    scopeGroups?: AccessToken['scopeGroups'];
+    enabledFeatures?: AccessToken['enabledFeatures'];
+    accountType?: AccessToken['accountType'];
+    authType?: AuthType;
+  };
+  diagnosis?: string;
+}
+
+const configFiles = [
```

Comment:
There are other files that can exist in projects too, like theme.json and cms-assets.json. I wonder though, rather than hardcode this list (which we'd have to continually update as new configs get added) could we instead just search the project for files with `.json` extensions and then run validation against all of them? The assumption being that any json file in the project is worth validating.

Reply:
Yeah, in the lack of a BE to provide these dynamically we kind of had to hardcode them 😞 

We could just validate all of the `json` files in the project if we wanted to.  That would mean that any user defined JSON files would also be validated, so it might be a lot in some of the larger projects.  But it scales with the addition of new files better.

Reply:
I changed it to validate all of the JSON files because it really doesn't add that much overhead.  I updated this list of have the two files you added so it adds them to the `configFiles` list which is just a QOL item for when users submit the output file